python_modules/libraries/dagster-dg/dagster_dg/docs.py

```diff
@@ -0,0 +1,168 @@
+            name, json_schema, subschema["anyOf"][0], available_scope=available_scope
+        )
+
+    objtype = subschema["type"]
+    if objtype == "object":
+        return CommentedObject(
+            key=name,
+            value={
+                k: _get_commented_obj(k, json_schema, v)
+                for k, v in subschema.get("properties", {}).items()
+            },
+            comment=f"Available scope: {available_scope}" if available_scope else None,
+        )
+    elif objtype == "array":
+        return [
+            _get_commented_obj(
+                name, json_schema, subschema["items"], available_scope=available_scope
+            )
+        ]
+    elif objtype == "string":
+        return "..."
+    elif objtype == "integer":
+        return 0
+    elif objtype == "boolean":
+        return False
+    else:
+        return f"<{objtype}>"
+
+
+class ComponentDumper(yaml.SafeDumper):
+    def increase_indent(self, flow=False, *args, **kwargs):
+        return super().increase_indent(flow=flow, indentless=False)
+
+    def write_line_break(self) -> None:
+        # add an extra line break between top-level keys
+        if self.indent == 0:
+            super().write_line_break()
+        super().write_line_break()
+
+    def _get_tag(self) -> str:
+        return getattr(self.event, "tag", "")
+
+    def expect_node(self, root=False, sequence=False, mapping=False, simple_key=False):
+        # for commented mappings, emit comment above the value
+        tag = self._get_tag()
+        if tag.startswith(COMMENTED_MAPPING_TAG):
+            self.write_indicator(f"# {tag[len(COMMENTED_MAPPING_TAG) + 1:]}", True)
+            self.write_line_break()
+
+        return super().expect_node(root, sequence, mapping, simple_key)
+
+    def process_tag(self):
+        tag = self._get_tag()
+        # ignore the mapping tag as it's handled specially
+        if tag.startswith(COMMENTED_MAPPING_TAG):
+            return
+        else:
+            super().process_tag()
+
+
+def commented_object_representer(dumper: yaml.SafeDumper, obj: CommentedObject) -> yaml.nodes.Node:
+    mapping = obj.value if isinstance(obj.value, dict) else {obj.key: obj.value}
+
+    if obj.comment is not None:
+        return dumper.represent_mapping(f"{COMMENTED_MAPPING_TAG}|{obj.comment}", mapping)
+    else:
+        return dumper.represent_dict(mapping)
+
+
+ComponentDumper.add_representer(CommentedObject, commented_object_representer)
+
+
+def generate_sample_yaml(component_type: str, json_schema: Mapping[str, Any]) -> str:
+    params_obj = _get_commented_obj("params", json_schema, json_schema)
+    return yaml.dump(
+        {"type": component_type, "params": params_obj}, Dumper=ComponentDumper, sort_keys=False
+    )
+
+
+def render_markdown_in_browser(markdown_content: str) -> None:
+    # Convert the markdown string to HTML
+    html_content = markdown.markdown(markdown_content)
+
+    # Add basic HTML structure
+    full_html = f"""
```

Comment:
might throw this in a template file?

Reply:
going to avoid doing that for now as I think this is not long for this world anyway